node/src/main/scala/org/bitcoins/node/PeerFinder.scala

```diff
@@ -130,17 +131,20 @@ case class PeerFinder(
             .filterNot(p => skipPeers().contains(p) || _peerData.contains(p))
 
           logger.debug(s"Trying next set of peers $peers")
-          peers.foreach(tryPeer)
+          val peersF = Future.sequence(peers.map(tryPeer))
+          Await.result(peersF, 10.seconds)
```

Comment:
Is there any benefit here to using `Await.result` compared to just mapping on `peersF` like so 

```scala 
peersF.failed.foreach(err => logger.error(s"Failed to connect to all peers,err"))
```

What is the significance of the `10.seconds` timeout? 

In my solution it doesn't block the thread at least, and I believe there is anything dependent on `peersF` being completed in 10 seconds?

Reply:
Nothing special about `10.seconds`, `Await` needed a duration and `10.second` is a generous one. But you are correct, there's nothing dependent on it.

As for why `Await`, the idea was that otherwise this block of code would be completed and scheduler will start the next countdown while things are still happening. With `Await`, the next countdown would only start once the future is done. `Runnable` run in their own thread so I felt using `Await` is good enough.

I'm, not really sure what would be better and to be fair, I think it doesn't really matter, both will do just fine. I will make this change, so that we are clearly sure that any `Await` that we have in our code base as far as `node` is concerned, is in an actor.